File: Benchmarking/Euler_11/euler11.py
import sys
import time
from time import clock
matLength = int(sys.argv[1])
numProd = int(sys.argv[2])

def makeData(matLength):
    string = "mat%d.txt" % matLength
    mat = [[0 for i in xrange(matLength)] for i in xrange(matLength)]

    with open(string) as f:
      j = 0
      for line in f:
        s = line.split()
        for i in range(0, len(s)):
          mat[j][i] = int(s[i])
        j += 1
      f.close()
    return mat

def a(mat, matLength, numProd):
  maxProd = 0

  for i in range(matLength):
    for j in range(matLength - numProd):
      #right/left
      prod = 1
      for k in range(0, numProd):
        prod = prod * mat[i][j+k]
      if prod > maxProd:
        maxProd = prod
      #up/down
      prod = 1
      for k in range(0, numProd):
        prod = prod * mat[j+k][i]
      if prod > maxProd:
        maxProd = prod

  #diagonal left->right
  for i in range(matLength - numProd):
    for j in range(matLength - numProd):
      prod = 1
      for k in range(0, numProd):
        prod = prod * mat[i + k][j + k]
      if prod > maxProd:
        maxProd = prod

  #diagonal right->left
  for i in range(numProd-1, matLength):
    for j in range(matLength-numProd):
      prod = 1
      for k in range(0, numProd):
        prod = prod * mat[i - k][j + k]
      if prod > maxProd:
        maxProd = prod

  return maxProd
#startData = time.clock()
data = makeData(matLength)
#endData = time.clock() - startData
#print endData
#startCalc = time.clock()

#a(data, matLength, numProd)

#endCalc = time.clock() - startCalc
#print endCalc

File: Benchmarking/Euler_11/euler11.java
import java.io.BufferedReader;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.concurrent.TimeUnit;

public class euler11
{
	public static void main (String[] args)
	{
		//a(Integer.parseInt(args[0]), Integer.parseInt(args[1]));
    int matLength = Integer.parseInt(args[0]);
    int numProd = Integer.parseInt(args[1]);

    long startData = System.currentTimeMillis();

    int[][] data = makeData(matLength);

    long endData = System.currentTimeMillis();
    float elapsedData = endData - startData;
    System.out.println(elapsedData / 1000);

    long startCalc = System.currentTimeMillis();

    System.out.println(a(data, matLength, numProd));

    long endCalc = System.currentTimeMillis();
    float elapsedCalc = endCalc - startCalc;
		System.out.println(elapsedCalc / 1000);
	}

  private static int[][] makeData(int matLength)
  {
    int[][] mat = new int[matLength][matLength];

    try (BufferedReader reader = Files.newBufferedReader(Paths.get("mat" + matLength + ".txt")))
    {
        int h = 0;
        String read;
        while((read = reader.readLine()) != null)
        {
            String[] words = read.split( "\\s+" );
            int v = 0;
            for (String s : words)
            {
                try
                {
                    int i = Integer.parseInt(s);
                    mat[h][v] = i;
                } catch (Exception e)
                {
                    System.out.println("File is corrupt");
                }
                v += 1;
            }
            h += 1;
        }
        reader.close();
    } catch (IOException ex)
    {
        System.out.println("File does not exist");
    }
    return mat;
  }

	private static long a(int[][] mat, int matLength, int numProd)
	{
				long maxProd = 0;
        long prod;
        for (int i = 0; i < matLength - numProd; i++)
        {
            for (int j = 0; j <= matLength - numProd; j++)
            {
                //Right-Left
                prod = 1;
                for (int k = 0; k < numProd; k++)
                {
                    prod *= mat[i + k][j];
                }
                if (prod > maxProd)
                {
                    maxProd = prod;
                }

                //Up-Down
                prod = 1;
                for (int k = 0; k < numProd; k++)
                {
                    prod *= mat[i][j + k];
                }
                if (prod > maxProd)
                {
                    maxProd = prod;
                }
            }
        }

        //Diagonal ->
        for (int i = 0; i <= matLength - numProd; i++)
        {
            for (int j = 0; j <= matLength - numProd; j++)
            {
                prod = 1;
                for (int k = 0; k < numProd; k++)
                {
                    prod *= mat[i + k][j + k];
                }
                if (prod > maxProd)
                {
                    maxProd = prod;
                }
            }
        }

        //Diagonal <-
        for (int i = 0; i <= matLength - numProd; i++)
        {
            for (int j = matLength - 1; j >= numProd - 1; j--)
            {
                prod = 1;
                for (int k = 0; k < numProd; k++)
                {
                    prod *= mat[i + k][j - k];
                }
                if(prod > maxProd)
                {
                    maxProd = prod;
                }
            }
        }

        return maxProd;
	}
}
